### PR TITLE
prometheus-modbus-exporter - new chart

### PR DESCRIPTION
**Pull request by openenergyprojects**
#### What this PR does / why we need it
this is a new chart. It focuses on modbus metrics. It deploys https://github.com/RichiH/modbus_exporter and creates prometheus serviceMonitors & more.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

#### Special notes for your reviewer

new chart, matching project: https://github.com/RichiH/modbus_exporter , which is maintained by @RichiH , @DaAwesomeP , @SuperQ and others.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by openenergyprojects** [comment]
@QuentinBisson  , @Xtigyro  and p8s admins.
We would appreciate your time to take a look over this and unblock the workflow.

File: charts/prometheus-modbus-exporter/README.md
Comment (QuentinBisson) [inline charts/prometheus-modbus-exporter/README.md:1]:
```suggestion
# prometheus-modbus-exporter

```

File: charts/prometheus-modbus-exporter/README.md
Comment (QuentinBisson) [inline charts/prometheus-modbus-exporter/README.md:5]:
```suggestion

This chart bootstraps a [modbus_exporter](https://github.com/RichiH/modbus_exporter) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.    
```

File: charts/prometheus-modbus-exporter/README.md
Comment (QuentinBisson) [inline charts/prometheus-modbus-exporter/README.md:7]:
```suggestion
The serviceMonitor objects are created for Prometheus Operator.    
```

File: charts/prometheus-modbus-exporter/templates/NOTES.txt
Comment (QuentinBisson) [inline charts/prometheus-modbus-exporter/templates/NOTES.txt:2]: Is that a valid command? Where should it be run?

File: charts/prometheus-modbus-exporter/templates/configmap.yaml
Comment (QuentinBisson) [inline charts/prometheus-modbus-exporter/templates/configmap.yaml:13]:
```suggestion
{{- end }}

```

File: charts/prometheus-modbus-exporter/templates/servicemonitor.yaml
Comment (QuentinBisson) [inline charts/prometheus-modbus-exporter/templates/servicemonitor.yaml:52]:
```suggestion
{{- end }}

```

File: charts/prometheus-modbus-exporter/templates/tests/test-connection.yaml
Comment (QuentinBisson) [inline charts/prometheus-modbus-exporter/templates/tests/test-connection.yaml:17]:
```suggestion
*/}}

```

**Comment by QuentinBisson** [comment]
I'm not sure I'm the best to add a new project here as i only recently became a maintainer. I think you should approach admins like @monotek @scottrigby or @torstenwalter

File: charts/prometheus-modbus-exporter/Chart.yaml
Comment (openenergyprojects) [reply charts/prometheus-modbus-exporter/Chart.yaml]: done, thanks!

**Comment by openenergyprojects** [comment]
many thanks @QuentinBisson  for the review!
I will wait for further comments from  @monotek @scottrigby or @torstenwalter or other admin.
I was wondering, shouldn't lint run before the approval flow?

**Comment by QuentinBisson** [comment]
@openenergyprojects I ran it for you

**Comment by openenergyprojects** [comment]
It seems the test env does not have Prometheus Operator CRDs installed, is this possible?
It does not know kind=ServiceMonitors

**Comment by QuentinBisson** [comment]
I think the other charts have it disabled by default 🤔 I'm not really sure how the pipeline is setup but I'll take a look

**Comment by openenergyprojects** [comment]
> I think the other charts have it disabled by default thinking I'm not really sure how the pipeline is setup but I'll take a look

that would be really strange. Anyway, for now I've conditioned the creation of serviceMonitor based on its presence: 
`{{ if .Capabilities.APIVersions.Has "monitoring.coreos.com/v1/ServiceMonitor" }}`

Let's retry the lint.

**Review by monotek — CHANGES_REQUESTED** [review]
please fix ci

**Review by monotek — CHANGES_REQUESTED** [review]
please also sign your commits, to get dco check passing...

File: charts/prometheus-modbus-exporter/Chart.yaml
Comment (monotek) [inline charts/prometheus-modbus-exporter/Chart.yaml:14]: please remove comments

File: charts/prometheus-modbus-exporter/values.yaml
Comment (monotek) [inline charts/prometheus-modbus-exporter/values.yaml:134]: please enable possible securitycontexts by default

File: charts/prometheus-modbus-exporter/values.yaml
Comment (monotek) [inline charts/prometheus-modbus-exporter/values.yaml:113]: please donß´t use latest but a real version number.

File: charts/prometheus-modbus-exporter/values.yaml
Comment (monotek) [inline charts/prometheus-modbus-exporter/values.yaml:103]: please dont use latest but a real versio number.
also it would be nice if the default tag would be the appVersion field form the chart.yaml

File: charts/prometheus-modbus-exporter/values.yaml
Comment (monotek) [inline charts/prometheus-modbus-exporter/values.yaml:64]: please define default values that work

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-modbus-exporter/values.yaml
Comment (openenergyprojects) [reply charts/prometheus-modbus-exporter/values.yaml]: Hi,
this is the plan, but for now the image maintained by @RichiH will have automatic builds (this is tracked here: (https://github.com/RichiH/modbus_exporter/issues/58)

File: charts/prometheus-modbus-exporter/values.yaml
Comment (openenergyprojects) [reply charts/prometheus-modbus-exporter/values.yaml:64]: without a running system (or modbus simulator), there are no working values.

File: charts/prometheus-modbus-exporter/values.yaml
Comment (openenergyprojects) [reply charts/prometheus-modbus-exporter/values.yaml]: updated

**Comment by openenergyprojects** [comment]
> please fix ci
Here I need some help to understand what is this error about:
```
2023-05-26 11:51:45 [INFO]   ERROR! Failed to call GitHub Status API!
2023-05-26 11:51:45 [INFO]   ERROR:[curl: (22) The requested URL returned error: 403]
2023-05-26 11:51:45 [ERROR]   ERRORS FOUND in MARKDOWN:[1]
2023-05-26 11:51:46 [INFO]   ERROR! Failed to call GitHub Status API!
2023-05-26 11:51:46 [INFO]   ERROR:[curl: (22) The requested URL returned error: 403]
2023-05-26 11:51:46 [INFO]   ERROR! Failed to call GitHub Status API!
2023-05-26 11:51:46 [INFO]   ERROR:[curl: (22) The requested URL returned error: 403]
2023-05-26 11:51:46 [FATAL]   Exiting with errors found!
```

**Comment by openenergyprojects** [comment]
> please also sign your commits, to get dco check passing...
Hmm, all commits are signed, not sure what is missing. As per the page here, it looks all are signed:
https://github.com/prometheus-community/helm-charts/compare/main...openenergyprojects:helm-charts:main

**Comment by monotek** [comment]
> > please fix ci
> > Hope is all good now. If not, please let me know what is more.
> 
> Here I need some help to understand what is this error about:
> 
> ```
> 2023-05-26 11:51:45 [INFO]   ERROR! Failed to call GitHub Status API!
> 2023-05-26 11:51:45 [INFO]   ERROR:[curl: (22) The requested URL returned error: 403]
> 2023-05-26 11:51:45 [ERROR]   ERRORS FOUND in MARKDOWN:[1]
> 2023-05-26 11:51:46 [INFO]   ERROR! Failed to call GitHub Status API!
> 2023-05-26 11:51:46 [INFO]   ERROR:[curl: (22) The requested URL returned error: 403]
> 2023-05-26 11:51:46 [INFO]   ERROR! Failed to call GitHub Status API!
> 2023-05-26 11:51:46 [INFO]   ERROR:[curl: (22) The requested URL returned error: 403]
> 2023-05-26 11:51:46 [FATAL]   Exiting with errors found!
> ```

you need to scroll up a bit from that message to see the actual error:

```
2023-05-27 19:07:24 [INFO]   File:[/github/workspace/charts/prometheus-modbus-exporter/README.md]
2023-05-27 19:07:24 [ERROR]   Found errors in [markdownlint] linter!
2023-05-27 19:07:24 [ERROR]   Error code: 1. Command output:
------
/github/workspace/charts/prometheus-modbus-exporter/README.md:22:55 MD034/no-bare-urls Bare URL used [Context: "https://github.com/ljean/modbu..."]
------
```

**Comment by monotek** [comment]
Please rebase in a way that not all old commits are put in this pr again.
I've set the pr to "draft" so not everybody is notified.

**Comment by openenergyprojects** [comment]
moved to: https://github.com/prometheus-community/helm-charts/pull/3531

**Pull request closed without merging**
